api-scanner/src/test/java/net/corda/plugins/AnnotatedClassesTest.java

```diff
@@ -0,0 +1,55 @@
+package net.corda.plugins;
+
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.BuildTask;
+import org.gradle.testkit.runner.GradleRunner;
+import org.gradle.testkit.runner.TaskOutcome;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class AnnotatedClassesTest {
+    @Rule
+    public TemporaryFolder testProjectDir = new TemporaryFolder();
+
+    @Before
+    public void setup() throws IOException {
+        File buildFile = testProjectDir.newFile("build.gradle");
+        CopyUtils.copyResourceTo("annotated-classes/build.gradle", buildFile);
+    }
+
+    @Test
+    public void testAnnotatedClasses() throws IOException {
+        BuildResult result = GradleRunner.create()
+            .withProjectDir(testProjectDir.getRoot())
+            .withArguments("scanApi", "--info")
+            .withPluginClasspath()
+            .build();
+        String output = result.getOutput();
+        System.out.println(output);
+
+        BuildTask scanApi = result.task(":scanApi");
+        assertNotNull(scanApi);
+        assertEquals(TaskOutcome.SUCCESS, scanApi.getOutcome());
+
+        File api = new File(testProjectDir.getRoot(), "build/api/annotated-classes.txt");
+        assertTrue(api.isFile());
+        assertEquals(
+            "@net.corda.example.NotInherited @net.corda.example.IsInherited public class net.corda.example.HasInheritedAnnotation extends java.lang.Object\n" +
```

Comment:
Asserting the equality of large texts like this makes the test brittle. Can we not at least split the output into a list of lines and assert that the list contains the bits we're looking for.

Reply:
The output _is_ supposed to be reproducible and constant. And I have tried to minimise the number of classes in each test...